File: IstTime.py
def convert_ist_time(ist_hour, ist_minute, offset):
    total_minutes = ist_hour * 60 + ist_minute + int(offset * 60)
    total_minutes %= 1440
    return divmod(total_minutes, 60)

time_differences = {
    "USA (New York)": -9.5,
    "UK (London)": -4.5,
    "Germany": -3.5,
    "Japan": 3.5,
    "Australia (Sydney)": 5.5,
    "UAE (Dubai)": -1.5,
    "China": 2.5
}

while True:
    print("\nEnter IST time (HH MM): ")
    ist_hour, ist_minute = map(int, input().split())

    print("\nChoose a country:")
    countries = list(time_differences.keys())
    for idx, country in enumerate(countries, 1):
        print(f"{idx}. {country}")

    choice = int(input("Enter choice number: ")) - 1
    selected_country = countries[choice]
    offset = time_differences[selected_country]

    new_hour, new_minute = convert_ist_time(ist_hour, ist_minute, offset)
    print(f"\nTime in {selected_country}: {new_hour:02}:{new_minute:02}")

    again = input("\nDo you want to use the program again? (yes/no): ").strip().lower()
    if again != "yes":
        print("Goodbye!")
        break

File: IstTime.java
import java.util.*;

public class IstTime {
    static Map<String, Double> timeDifferences = new HashMap<>();

    static {
        timeDifferences.put("USA (New York)", -9.5);
        timeDifferences.put("UK (London)", -4.5);
        timeDifferences.put("Germany", -3.5);
        timeDifferences.put("Japan", 3.5);
        timeDifferences.put("Australia (Sydney)", 5.5);
        timeDifferences.put("UAE (Dubai)", -1.5);
        timeDifferences.put("China", 2.5);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String again;

        do {
            System.out.print("Enter IST time in HH:mm (24-hour format): ");
            String inputTime = scanner.nextLine();

            System.out.println("Choose a country:");
            int i = 1;
            List<String> countries = new ArrayList<>(timeDifferences.keySet());
            for (String country : countries) {
                System.out.println(i + ". " + country);
                i++;
            }

            int choice = scanner.nextInt();
            scanner.nextLine(); // consume newline
            String selectedCountry = countries.get(choice - 1);
            double offset = timeDifferences.get(selectedCountry);

            String[] parts = inputTime.split(":");
            int hours = Integer.parseInt(parts[0]);
            int minutes = Integer.parseInt(parts[1]);

            int totalMinutes = (int)(hours * 60 + minutes + offset * 60);
            if (totalMinutes < 0) totalMinutes += 1440;
            totalMinutes %= 1440;

            int newHours = totalMinutes / 60;
            int newMinutes = totalMinutes % 60;

            System.out.printf("Time in %s: %02d:%02d\n", selectedCountry, newHours, newMinutes);

            System.out.print("Do you want to use the program again? (yes/no): ");
            again = scanner.nextLine().trim().toLowerCase();
        } while (again.equals("yes"));

        System.out.println("Goodbye!");
    }

}
 
